### PR TITLE
feat(radio):add unique id

### PR DESCRIPTION
**Pull request by jiaoy14**
Fixes: https://github.com/uber/baseweb/issues/3411

### CODE REVIEW

**Comment by chasestarr** [comment]
The change in the `.scenario.js` file requires an edit in https://github.com/uber/baseweb/blob/master/src/radio/__tests__/radio.e2e.js#L29-L30. There is now an additional `label` element than previously expected.

```js
it('changes selection on radio click', async () => {
    await mount(page, 'radio');
    expect(await checkedByValue(2)).toBeTruthy();
    // Changes the two lines below
    await page.click('label:nth-of-type(3)');
    expect(await checkedByValue(3)).toBeTruthy();
    expect(await checkedByValue(2)).toBeFalsy();
  });
```

**Comment by jiaoy14** [comment]
''' JavaScript

    expect(await checkedByValue(3)).toBeTruthy();
    await page.click('label:nth-of-type(2)');
    expect(await checkedByValue(2)).toBeTruthy();
    expect(await checkedByValue(3)).toBeFalsy();
 
'''

File: src/radio/__tests__/radio.e2e.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ const {mount, analyzeAccessibility} = require('../../../e2e/helpers');
 
 async function checkedByValue(n) {
   const input = await page.$(`input[value="${n}"]`);
+  console.log(input);
```
Comment (tajo) [inline src/radio/__tests__/radio.e2e.js:15]:
```suggestion
```

Comment (jiaoy14) [reply src/radio/__tests__/radio.e2e.js]: Yeah, you are correct!

Comment (jiaoy14) [reply src/radio/__tests__/radio.e2e.js]: Yeah,you are correct! Many thanks!

**Pull request merged**